runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -2353,3 +2354,186 @@ J9::TransformUtil::specializeInvokeExactSymbol(TR::Compilation *comp, TR::Node *
       }
       return false;
    }
+
+bool
+J9::TransformUtil::refineMethodHandleInvokeBasic(TR::Compilation* comp, TR::TreeTop* treetop, TR::Node* node, TR::KnownObjectTable::Index mhIndex, bool trace)
+   {
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   auto knot = comp->getKnownObjectTable();
+   if (mhIndex == TR::KnownObjectTable::UNKNOWN ||
+       knot->isNull(mhIndex))
+      {
+      if (trace)
+         traceMsg(comp, "MethodHandle for invokeBasic n%dn %p is unknown or null\n", node->getGlobalIndex(), node);
+      return false;
+      }
+
+   TR_J9VMBase* fej9 = comp->fej9();
+   auto targetMethod = fej9->targetMethodFromMethodHandle(comp, mhIndex);
```

Comment:
In `targetMethodFromMethodHandle`, from KOT we get the object at the given index and from the object we get a j9method. Can the the mhIndex <--> j9method association ever change? If not, I am thinking of caching this association to avoid extra messages for JITServer. FYI @dmitry-ten

Reply:
That's a good question, it seems to me that the j9method of a MemberName can change due to HCR #11582. @babsingh Can you confirm?

Reply:
Yes, it can change due to HCR. HCR support for OJDK MHs is being tracked through https://github.com/eclipse/openj9/issues/11528.

Reply:
Since KOT only exists during a compilation we should be fine because class redefinition aborts compilations in progress.

Reply:
Yes, HCR will be addressed in #11528, and the ultimate goal is to update MemberName fields in redefinition event so that we don't get stale j9method pointer